python_configure.bzl

```diff
@@ -441,3 +443,62 @@ Args:
       will configure for that Python version instead of the installed
       binary. This configuration takes precedence over python_version.
 """
+
+def _parse_my_own_version_from_module_dot_bazel(ctx):
+    lines = ctx.read(Label("//:MODULE.bazel")).split("\n")
+    for line in lines:
+        parts = line.split("\"")
+        if parts[0] == "    version = ":
+            return parts[1]
+    _fail("Failed to parse my own version from `MODULE.bazel`! " +
+          "This should never happen!")
+
+def _extension_impl(ctx):
+    toolchain = None
+    for module in ctx.modules:
+        if module.is_root:
+            if not module.tags.toolchain:
+                pass
+            elif len(module.tags.toolchain) == 1:
+                toolchain = module.tags.toolchain[0]
+            else:
+                _fail("The root module may not specify multiple `toolchain` " +
```

Comment:
Sorry I'm a bit late. But I think we can avoid throwing such errors but try to resolve the toolchains defined by
1) if the root module specifies multiple one, respect the first one and print a warning message.
2) if non-root module specifies toolchains, then just ignore it, throwing an error won't be very helpful since the user cannot change the root module easily.

I think this will make it more user friendly.